apm-sniffer/apm-toolkit-activation/apm-toolkit-logback-1.x-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/log/logback/v1/x/mdc/PrintMDCTraceIdInterceptor.java

```diff
@@ -36,7 +36,11 @@
 
     @Override public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
         Class<?>[] argumentsTypes, Object ret) throws Throwable {
-        return "TID:" + ContextManager.getGlobalTraceId();
+        String tid = ContextManager.getGlobalTraceId();
+        if ("N/A".equals(tid)) {
+            tid = (String) ((EnhancedInstance) allArguments[0]).getSkyWalkingDynamicField();
+        }
```

Comment:
Why couldn't getDynamicField first? Then call `ContextManager#getGlobalTraceId`, should be more efficient.

Reply:
Sorry, I forgot to update here, this is what we discussed at the time, because objInst is an interface, there may be multiple implementations, but we just enhanced `ch.qos.logback.classic.spi.LoggingEvent`, so use `ContextManager .isActive()` is true must be synchronous, when there is no trace, the speculative asynchronous decision logic